### PR TITLE
stabilize collections_range (closes #30877)

### LINKED ISSUE #30877: Tracking issue for RangeBounds
This issue tracks the `collections_range` feature which applies to `std::collections::range::RangeArgument`. It used to be lumped in with `drain` (#27711) but that was stabilized while `RangeArgument` may continue to be affected by experiments with inclusive ranges, so here we are.

### PR DESCRIPTION
**Pull request by budziq**
This is my first PR so excuse me if I have made some mistakes.
I know that similar PR's for stabilizing this feature have been reverted in the past.

### CODE REVIEW

File: src/liballoc/range.rs [hunk 1]
```diff
@@ -19,6 +16,7 @@ use Bound::{self, Excluded, Included, Unbounded};
 
 /// `RangeArgument` is implemented by Rust's built-in range types, produced
 /// by range syntax like `..`, `a..`, `..b` or `c..d`.
+#[stable(feature = "collections_range", since = "1.21.0")]
```
Comment (kennytm) [inline src/liballoc/range.rs]: Unless this is meant to be backport to beta, the `since` should be 1.22.0

Comment (budziq) [reply src/liballoc/range.rs]: Woops! Fixed. thanks @kennytm!

**Review by dtolnay — CHANGES_REQUESTED** [review]
Thanks but from what I can tell, this is not ready to be stabilized yet.

The tracking issue is pretty confusing right now. It would be really helpful if you could look through all the RangeArgument-related work that has happened since the start of the final comment period, and synthesize a detailed summary of where we stand and what people have been asking to change. That includes at least the following threads:

- https://github.com/rust-lang/rust/issues/30877
- https://github.com/rust-lang/rust/pull/42991
- https://github.com/rust-lang/rust/pull/43033
- https://github.com/rust-lang/rust/pull/44518

**Comment by dtolnay** [comment]
I wrote a summary in https://github.com/rust-lang/rust/issues/30877#issuecomment-330014651. This API is being actively worked on in #44518 so I cancelled the FCP.

**Comment by budziq** [comment]
Argh sorry! I've procrastinated too long with this one. Thanks for taking this up @dtolnay!

**Pull request closed without merging**